Feat: new editing widget for popup cells(Boolean, Date, DateTime, Select and Time)

DMouayad opened the pull request:
## Demo

### TrinaBooleanCell

https://github.com/user-attachments/assets/51b5fabc-9147-466a-8dce-bfc1e0bb5e7c

### TrinaDateCell

https://github.com/user-attachments/assets/4d5e6e01-b4dd-4e49-a250-8ac365c95043


### TrinaDateTimeCell


https://github.com/user-attachments/assets/8f27531e-a453-4457-a43d-2fbe5fd2540a


### TrinaSelectCell


https://github.com/user-attachments/assets/5f5285de-5adf-47c0-8c30-34bc6da0bb3d



### TrinaTimeCell


https://github.com/user-attachments/assets/0b6c31b3-5310-4bb6-bb6d-035187ab7651


## New Widgets

**ONLY used internally**

### TrinaTimePicker

A customizable time picker widget with hour and minute input fields, validation and keyboard shortcuts support.

### TrinaDatePicker

A wrapper around Flutter's CalendarDatePicker

### TrinaDropdownMenu

A customizable select menu using `MenuAnchor` with filtering and search.

### TrinaPopup

A flexible popup widget that displays content anchored to a target.


## API changes

### Breaking changes

* `TrinaColumnType.select`
    * `onItemSelected` callback changed: It now provides the selected item directly, instead of a
      TrinaGridOnSelectedEvent.
    * `width` parameter renamed to `menuWidth`.
    * `builder` parameter renamed to `menuItemBuilder`.

* `TrinaColumnType.boolean`
    * `onItemSelected` callback changed: It now provides the selected item directly, instead of a
      TrinaGridOnSelectedEvent.
    * `width` parameter renamed to `menuWidth`.
    * `builder` parameter renamed to `menuItemBuilder`.

###  Non-Breaking Changes (New Features & Enhancements)

* `TrinaColumnType.select`
    * New parameters added: menuItemHeight, menuMaxHeight, itemToString, and itemToValue for more control
      over the dropdown menu.

* New factory `TrinaColumnType.selectWithSearch`: Provides a dropdown with a search field.

* New factory `TrinaColumnType.selectWithFilters`: provides a dropdown with advanced, customizable filters.

* `TrinaColumnType.date`
    * New parameter added: closePopupOnSelection to control the behavior of the date picker after a selection
      is made.

* `TrinaColumnType.time`
    * New parameters added: autoFocusMode, saveAndClosePopupWithEnter, minTime, and maxTime for more control
      over the time picker.

## Tests

- Added tests for `TrinaTimePicker`, covering input, validation, keyboard navigation, and callback functionality.
- Added tests for `TrinaDateTimeCell`.
- Refactored tests for date, select, and time cells to cover new UI.

### Related

- resolve #27

doonfrs commented:
Wow @DMouayad, that was done really well!

I found a few bugs:

1. I'm using keyboard input only (as most users will). Pressing space opens the popup, which is great. But after some edits, I got this error on boolean cells:


```
[ERROR:flutter/runtime/dart_vm_initializer.cc(40)] Unhandled Exception: A TextEditingController was used after being disposed.
Once you have called dispose() on a TextEditingController, it can no longer be used.
#0      ChangeNotifier.debugAssertNotDisposed.<anonymous closure> (package:flutter/src/foundation/change_notifier.dart:182:9)
change_notifier.dart:182

```

2. On mobile, there's a layout issue:
The last cell (on the right or bottom) doesn't show the calendar dropdown.
<img width="779" height="840" alt="image" src="https://github.com/user-attachments/assets/e5539951-b2fa-44cf-af58-298ab994f6e6" />


3. Selecting a date from the calendar should automatically close the popup. Currently, I still need to click the OK button, which feels unnecessary.

doonfrs commented:
@DMouayad 
once you publish the updates, I will re-test and try to repreduce the keyboard error

DMouayad commented:
We can also make it configurable i.e. add a boolean flag `selectWithSingleTap` or `closePopupOnSelection` or `shouldClosePopupOnSelection`.

doonfrs commented:
@DMouayad Please find the contributers list in the codebase and add your name / github handle.

DMouayad commented:
@doonfrs  Finally this's done.

### New things since the first iteration:

- Select menu has been re-designed both in UI and in code.
- Time picker UI has been improved.
- Fixed multiple issues and bugs.
- Added additional tests for new components.

### Notes

- It has some small breaking changes. so a major version is needed.
- I suggest publishing it alone (i.e. with no other commits).
- If you don't want to publish this at the moment we can create a feature branch and we merge those commits to it (I'll create another PR) or we delay the merge. It's up to you.
- THank u for ur time.

doonfrs commented:
Thank you @DMouayad , a really great work!
That was the biggest and the most usefull update to the grid.
I deployed the latest feature to 2.0.0

Also create the contributers page:
https://github.com/doonfrs/trina_grid/blob/main/doc/contributing/contributors.md

DMouayad commented:
one small note: I should have explained more why  
> I suggest publishing it alone (i.e. with no other commits).
Because  if it's released with other bug fixes and those fixes breaks smth else then the user will have to revert to a previous version and can't use the new feature.
